graphene/src/document.rs

```diff
@@ -41,6 +54,15 @@ impl Document {
 		self.hasher.finish()
 	}
 
+	pub fn serialize_document(&self) -> String {
+		let doc = serde_json::json!({
+			"version": 0,
+			"root": self.root,
+			"nodes": {}
+		});
```

Comment:
This is the main "format". Open question whether this is "good enough" or we want to think of a format with better backwards compatibility and security.

Reply:
I guess the format will still change a bit before 0.1. I don't think it is feasible to aim for creating a backwards compatible format, but rather providing a script to convert files to the newest version. (If we e.g. rename/add/delete fields from Layer)